### PR TITLE
Feat: Listen for POST Request from Frontend 

### PR DESCRIPTION
**Pull request by chaoyue-zhao**
**PROBLEM**
Backend needs to listen for POST request from front end with information regarding the poll submitted. 

**SOLUTION**
- Created new-poll route and added to app
- *Super-Important* Added body-parser to parse the request body
- Processed POST request in controller

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
 
+
 // Create a new express application instance
 const app = express();
 
 // The port the express app will listen on
 const port = process.env.PORT;
-
-logger.info("🤖 Initializing middleware");
+logger.info(":robot_face: Initializing middleware");
```
Comment (MustafaHaddara) [inline src/index.js]: lol does your terminal render this properly?

File: src/routes/new-poll/new-poll.controller.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const newPolls = function (req, res) {
+  const body = req.body;
+  console.log(body);
+  res.set('Content-Type', 'application/json')
+  res.send(`Received new Poll: ${body.message}`);
```
Comment (MustafaHaddara) [inline src/routes/new-poll/new-poll.controller.js]: 👍

File: src/index.js [hunk 3]
```diff
@@ -37,14 +41,16 @@ app.use(
...
+
+// Add channels 
 // app.use("/", router);
 // app.use(errorHandler);
+app.use(bodyParser());
 app.use("/channels", channelRouter);
 // // app.get("/", );
-
+app.use("/new-poll", newPollRouter);
 app.use("/previous-polls", previousPollsRouter);
```
Comment (MustafaHaddara) [inline src/index.js]: this is totally a style thing, my personal preference would be to expose the following two routes:

```
GET /polls
POST /polls
```

the GET would do what your `previous-polls` route is currently doing, and the POST would do what your `new-polls` route is doing. 

This brings the app more inline with common RESTful practices...they're referring to the same resource, the differentiator is in the HTTP method (or "verb") used to access them. 

However, this is totally up to you, what you have here will work too!

File: src/index.js (hunk 1 above)
Comment (chaoyue-zhao) [reply src/index.js]: Oh my. That's why the emojis are not showing.

File: src/index.js (hunk 3 above)
Comment (chaoyue-zhao) [reply src/index.js]: Aha, that actually makes more sense and also make the structure dry. I have made the change. Thank you!

**Pull request merged**